frontend/src/tests/application-tests.js

```diff
@@ -1,48 +1,172 @@
+            session_id: session.id,
+        };
+        let resp = await apiPOST("/admin/postings", posting);
+        expect(resp).toHaveStatus("success");
+        return resp.payload;
+    }
+
+    beforeAll(async () => {
+        await databaseSeeder.seed({ apiGET, apiPOST });
+        session = databaseSeeder.seededData.session;
+        applicant = databaseSeeder.seededData.applicant;
+        position = databaseSeeder.seededData.position;
+
+        // Get the current active user
+        let resp = await apiGET("/debug/active_user");
+        expect(resp).toHaveStatus("success");
+        adminUser = resp.payload;
+
+        // Making sure the user is the seeded applicant before this get request for the purposes of the next test
+        // Created a user for the applicant to make sure they exist before changing the default user to applicant
+        Object.assign(userCreatedFromApplicant, {
+            utorid: applicant.utorid,
+            roles: ["ta"],
+        });
+
+        resp = await apiPOST("/debug/users", userCreatedFromApplicant);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userCreatedFromApplicant, resp.payload);
+
+        // Create a new user with TA permissions
+        resp = await apiPOST("/debug/users", userWithTaPermissions);
+        expect(resp).toHaveStatus("success");
+        Object.assign(userWithTaPermissions, resp.payload);
+
+        // Create a new posting for PF tests
+        const respPostingData = await addPosting(
+            postingDataForPFTests,
+            session
+        );
+        Object.assign(postingDataForPFTests, respPostingData);
+    }, 30000);
+
     // These tests set data through the `/public/postings` route,
     // but read data through the `/api/v1/admin` route.
     describe("Public route tests", () => {
-        beforeAll(async () => {
-            await databaseSeeder.seed({ apiGET, apiPOST });
-            session = databaseSeeder.seededData.session;
-            position = databaseSeeder.seededData.position;
-            let resp = await apiGET(`/debug/active_user`);
+        it("Get survey.js posting data through public route", async () => {
+            // Make sure the user has admin permissions before this post request
+            // Make a new posting and update <posting> to include the id of the posting
+            const respPostingData = await addPosting(postingData, session);
+            checkPropTypes(postingPropTypes, respPostingData);
+            expect(respPostingData.id).not.toBeNull();
+            Object.assign(postingData, respPostingData);
+
+            await switchToUser(userCreatedFromApplicant);
+
+            // Read survey.js posting data
+            let resp = await apiGET(
+                `/public/postings/${postingData.url_token}`,
+                true
+            );
             expect(resp).toHaveStatus("success");
-            adminUser = resp.payload;
+            checkPropTypes(surveyPropTypes, resp.payload);
```

Comment:
Where did you check that the payload actually has all the data that a posting should?